test/powershell/Modules/Microsoft.PowerShell.Management/Registry.Tests.ps1

```diff
@@ -3,40 +3,43 @@ Describe "Basic Registry Provider Tests" -Tags @("CI", "RequireAdminOnWindows")
         $restoreLocation = Get-Location
         
         #skip all tests on non-windows platform
-        $originalDefaultParameterValues = $PSDefaultParameterValues.Clone()
-        if ($IsWindows -eq $false) {
+        $defaultParamValues = $PSdefaultParameterValues.Clone()
+        if (!$IsWindows) {
             $PSDefaultParameterValues["it:skip"] = $true
-            return
+        } else {
+            $registryBase = "HKLM:\software\Microsoft\PowerShell\3\"
+            $parentKey = "TestKeyThatWillNotConflict"
+            $testKey = "TestKey"
+            $testKey2 = "TestKey2"
+            $testPropertyName = "TestEntry"
+            $testPropertyValue = 1
         }
-
-        $registryBase = "HKLM:\software\Microsoft\PowerShell\3\"
-        $parentKey = "TestKeyThatWillNotConflict"
-        $testKey = "TestKey"
-        $testKey2 = "TestKey2"
-        $testPropertyName = "TestEntry"
-        $testPropertyValue = 1
     }
 
     AfterAll {
         #restore the previous environment
         Set-Location -Path $restoreLocation
```

Comment:
This is one of the spots I had in mind when I mentioned "more complete grouping of actions...". A principle I'd like the tests to follow: if a Describe is being skipped, it should not do anything. In this case, it is saving and restoring the location; an action that can be bypassed with `if ($IsNotSkipped)`